engine/schema/src/com/cloud/network/vpc/dao/VpcServiceMapDaoImpl.java

```diff
@@ -62,7 +62,31 @@ protected VpcServiceMapDaoImpl() {
 
     @Override
     public boolean areServicesSupportedInVpc(long vpcId, Service... services) {
-        // TODO Auto-generated method stub
+        SearchCriteria<VpcServiceMapVO> sc = MultipleServicesSearch.create();
+        sc.setParameters("vpcId", vpcId);
+
+        if (services != null) {
+            String[] servicesStr = new String[services.length];
+
+            int i = 0;
+            for (Service service : services) {
+                servicesStr[i] = service.getName();
+                i++;
+            }
+
+            sc.setParameters("service", (Object[])servicesStr);
+        }
+
+        List<VpcServiceMapVO> vpcServices = listBy(sc);
+
+        if (services != null) {
+            if (vpcServices.size() == services.length) {
```

Comment:
listBy is going to query the services that are send to the function. So if all the services queried for are present the dao should return that many rows.